pkg/sources/fake/fake.go

```diff
@@ -14,122 +15,154 @@ const (
 	readInterval = 400 * time.Millisecond
 )
 
-const (
-	// TODO(karlding): change these to the actual values
-	rightTurnOn = iota
-	rightTurnOff
-	leftTurnOn
-	leftTurnOff
-	hazardOn
-	hazardOff
-	solarPowerLevel
-	motorPowerLevel
-	batteryState
-	cruiseOn
-	cruiseLevel
-	cruiseOff
-	speed
-)
-
 var startTime = time.Now()
 
+var solarFront = false
 var rightTurnState = false
 var leftTurnState = false
 var hazardState = false
 var cruiseControlState = false
 
-func getCAN(id uint32) msgs.CAN {
-	t := time.Since(startTime)
-	fid := float64(id)
-	data := uint64(800 + 150*math.Sin((t.Seconds()/20)+fid))
+// Utility functions to generate a 64-bit integer by bitshifting fewer-bit integers
+// The 64-bit raw data returned by these functions is big-endian
+// (e.g. first value is rightmost in 64 bits, last value is leftmost in 64 bits)
+func createDataU8(values ...uint8) uint64 {
+	data := uint64(0)
+	for index, value := range values {
+		data = (data) | uint64(value)<<(uint(index)*8)
+	}
+	return data
+}
 
-	return msgs.NewCAN(id, data, 0)
+func createDataU16(values ...uint16) uint64 {
+	data := uint64(0)
+	for index, value := range values {
+		data = (data) | uint64(value)<<(uint(index)*16)
+	}
+	return data
+}
+
+func createDataU32(values ...uint32) uint64 {
+	data := uint64(0)
+	for index, value := range values {
+		data = (data) | uint64(value)<<(uint(index)*32)
+	}
+	return data
 }
 
 // GenFake generates fake data
 func GenFake(bus *pubsub.MessageBus) {
-	batteryValue := uint64(0)
 	rand, _ := randutil.NewPseudoRand()
 	for {
 		// motor power
-		getCAN(motorPowerLevel)
-		bus.Publish("CAN", getCAN(motorPowerLevel))
+		if uint64(randutil.RandIntInRange(rand, 0, 5)) == 0 {
+			mcVoltage1 := uint16(randutil.RandIntInRange(rand, 0, 70))
```

Comment:
Note that you'll probably want to change these values if the goal is to provide realistic data that reflects how the car will behave.

But this works, so I'm not going to look too hard at all these ranges! :)

Reply:
(Also might be worth defining the constants for the upper/lower bounds for all of these values, so these aren't magic numbers)